File: test-suite/src/test/java/io/micronaut/views/docs/soy/SoyRenamingController.java
package io.micronaut.views.docs.soy;

import io.micronaut.context.annotation.Requires;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.views.View;

import java.util.Collections;
import java.util.Map;

@Requires(property = "spec.name", value = "soy")
@Controller("/soy/renaming")
public class SoyRenamingController {

    @View("sample.renaming")
    @Get
    public Map<String, Object> renaming() {
        return Collections.emptyMap();
    }
}

File: test-suite-python/src/test/python/micronaut/views/docs/soy/SoyRenamingController.py
from micronaut.context.annotation import Requires
from micronaut.http.annotation import Controller, Get
from micronaut.views import View


@Requires(property="spec.name", value="soy")
@Controller("/soy/renaming")
class SoyRenamingController:

    @View("sample.renaming")
    @Get
    def renaming(self) -> dict[str, object]:
        return {}
